File: app-java/src/main/java/com/java/bilibili/utils/FileUtil.java
package com.java.bilibili.utils;

import java.io.File;
import java.io.FileOutputStream;

/**
 * Created by Eric.zhang on 2018/1/10.
 */

public class FileUtil {
    public static File getDirectory(String path) {
        File appDir = new File(path);
        if (!appDir.exists()) {
            appDir.mkdirs();
        }
        return appDir;
    }
    public static int writeFile(String path, String content) {
        try {
            File f = new File(path);
            if (f.exists()) {
                f.delete();
            }
            if (f.createNewFile()) {
                FileOutputStream utput = new FileOutputStream(f);
                utput.write(content.getBytes());
                utput.close();
            } else {
                return 0;
            }
        } catch (Exception e) {
            return 0;
        }
        return 1;
    }
}

File: app/src/main/java/com/kotlin/bilibili/utils/FileUtil.kt
package com.kotlin.bilibili.utils

import java.io.File
import java.io.FileOutputStream


object FileUtil {

    fun getDirectory(path: String): File {
        val appDir = File(path)
        if (!appDir.exists()) {
            appDir.mkdirs()
        }
        return appDir
    }

    fun writeFile(path: String, content: String): Int {
        try {
            val f = File(path)
            if (f.exists()) {
                f.delete()
            }
            if (f.createNewFile()) {
                val utput = FileOutputStream(f)
                utput.write(content.toByteArray())
                utput.close()
            } else {
                return 0
            }
        } catch (e: Exception) {
            return 0
        }
        return 1
    }
}
